### PR TITLE
optimize(balancer) faster creation of a randomly ordered list

### PR DESCRIPTION
**Pull request by hishamhm**

### CODE REVIEW

File: src/resty/dns/balancer.lua [hunk 1]
```diff
@@ -982,6 +982,33 @@ function objBalancer:setCallback(callback)
   self.callback = callback
 end
 
+local randomlist_cache = setmetatable({}, { __mode = "k" })
```
Comment (Tieske) [inline src/resty/dns/balancer.lua]: why the weak-table here? since it is indexed by a scalar (the size) it will never be removed anyway, right?

Comment (hishamhm) [reply src/resty/dns/balancer.lua]: Ah yes, I changed that but forgot to commit

Comment (hishamhm) [reply src/resty/dns/balancer.lua]: In any case, the upper bound of memory use for this table is pretty low so we can just keep this cache around.

Comment (Tieske) [reply src/resty/dns/balancer.lua]: yes, size shouldn't be an issue. But, as I'm writing this, wondering, maybe add a method to clear the cache? It usually keeps annoyingly lingering in the back of my head that there is a potential memory-leak somewhere, so maybe better to make it explicit by adding a method to clear the cache? What's your take?

Comment (hishamhm) [reply src/resty/dns/balancer.lua]: It's not a leak if we know exactly how much memory we're spending at most. :) Consider it "base memory requirements" of the program and sleep well at night. Adding a method that will never be called in practice adds code, doc and testing overhead. And in practice, how many different slot sizes will ever be used? We're talking negligible memory use in real scenarios.

Comment (Tieske) [reply src/resty/dns/balancer.lua]: k, let's do this then.

**Pull request merged**